lms/services/course.py

```diff
@@ -67,7 +67,7 @@ def _is_pre_sections(self, authority_provided_id):
 
 def course_service_factory(_context, request):
     return CourseService(
-        request.find_service(name="ai_getter"),
+        request.find_service(name="application_instance").get(),
```

Comment:
This means that _constructing_ the `CourseService` can now raise `ConsumerKeyError`, whereas previously it wouldn't have been raised until `get_or_create()` was called. `CourseService` is used in a few places, I've had a look through them all and I don't think this will make any difference.

Reply:
Changed this back. I don't think it's a good idea for `CourseService.__init__()` to raise as that would cause `request.find_service(name="course")` to raise which I think would be unexpected and a likely source of errors: https://github.com/hypothesis/lms/pull/2651/commits/6ad80f0e01c485252a3c2fc52d502bd7afb2156f